src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -388,16 +387,46 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
             if (string.IsNullOrEmpty(validIssuerTemplate))
                 return false;
 
-            if (validIssuerTemplate.Contains(TenantIdTemplate))
+            int indexOfTenantIdTemplate = validIssuerTemplate.IndexOf(TenantIdTemplate, StringComparison.Ordinal);
+            if (indexOfTenantIdTemplate >= 0)
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate.AsSpan(), TenantIdTemplate.AsSpan(), indexOfTenantIdTemplate, actualIssuer.AsSpan(), tenantId.AsSpan());
             }
             else
             {
                 return validIssuerTemplate == actualIssuer;
             }
         }
 
+        /// <summary>
+        /// Compare two Issuers with templates without string allocations.
+        /// This function replaces: issuer1.Replace(issuer1Template, tenantId) == issuer2
+        /// Example:
+        ///     issuer1 = "https://login.microsoftonline.com/{tenantid}/v2.0"
+        ///     issuer1Template = "{tenantid}"
+        ///     issuer2 = "https://login.microsoftonline.com/12345678/v2.0"
+        ///     tenantId = "12345678"
+        /// </summary>
+        internal static bool IssuersWithTemplatesAreEqual(ReadOnlySpan<char> issuer1, ReadOnlySpan<char> issuer1Template, int templateStartIndex, ReadOnlySpan<char> issuer2, ReadOnlySpan<char> tenantId)
+        {
+            var issuer2TemplateStartIndex = issuer2.IndexOf(tenantId);
+            if (templateStartIndex == -1 || issuer2TemplateStartIndex == -1 || templateStartIndex != issuer2TemplateStartIndex)
+                return false;
+
+            // ensure the first part of the issuer1 matches the first part of issuer2
+            if (issuer1.Slice(0, templateStartIndex).SequenceCompareTo(issuer2.Slice(0, templateStartIndex)) != 0)
```

Comment:
Why not SequenceEquals with a StringComparer?

Reply:
Ah yes, we don't need to know their relative order, so SequenceEquals is better. 😄 Changing to SequenceEquals